index.js

```diff
@@ -1,106 +1,151 @@
 const path = require("path");
 const { execSync } = require("child_process");
 const fs = require("fs");
-const axios = require('axios');
-const FormData = require('form-data');
-const converter = require('json-2-csv');
-const split = require('split');
-const { finished } = require('stream');
+const axios = require("axios");
+const FormData = require("form-data");
+const converter = require("json-2-csv");
+const split = require("split");
+const { finished } = require("stream");
 // Constructing promisify from util
-const { promisify } = require('util');
+const { promisify } = require("util");
 // Defining finishedAsync method
 const finishedAsync = promisify(finished);
 
+const eventEndTimeLimit = 5 * 60 * 1000; // after this time limit, events are sent even if not ended
+
 function initialize(config, logger) {
-    this.config = config;
-    this.logger = logger;
-    this.config.logFilename = 'active.log';
+  this.config = config;
+  this.logger = logger;
+  this.config.logFilename = "active.log";
 }
 
 async function aggregateTelemetryData() {
-    this.csvFilename = 'telhub_logs.csv'
-    const otNodeLogsPath = path.join(this.config.appRootPath, 'logs');
-    const intermediateConversionFile = path.join(otNodeLogsPath, `intermediateFile.log`);
+  this.csvFilename = "telhub_logs.csv";
+  const otNodeLogsPath = path.join(this.config.appRootPath, "logs");
+  const intermediateConversionFile = path.join(
+    otNodeLogsPath,
+    `intermediateFile.log`
+  );
+
+  this.logger.info("Started sending telemetry data command");
+
+  try {
+    execSync(
+      `cat ${path.join(
+        otNodeLogsPath,
+        this.config.logFilename
+      )} | grep \'"level":15\' | grep -v \'level-change\' > ${intermediateConversionFile}`
+    );
+  } catch (e) {
+    // No data to be returned
+    return null;
```

Comment:
Check if log file is empty before parsing (cat) is called

Reply:
just additional info, this error will be thrown only if there are no level 15 events, so checking if file is empty won't prevent this